nifi-nar-bundles/nifi-extension-utils/nifi-event-transport/src/test/java/org/apache/nifi/event/transport/netty/SocketByteArrayMessageDecoderTest.java

```diff
@@ -0,0 +1,166 @@
+    @Mock
+    private ChannelOutboundHandlerAdapter nonSslHandler;
+    @Mock
+    private SSLEngine sslEngine;
+    @Mock
+    private SSLSession sslSession;
+
+    @BeforeEach
+    public void init() {
+        inetSocketAddress = new InetSocketAddress("localhost", 21000);
+        Mockito.when(channelHandlerContext.channel()).thenReturn(channel);
+        Mockito.when(channel.remoteAddress()).thenReturn(inetSocketAddress);
+        Mockito.when(channel.pipeline()).thenReturn(channelPipeline);
+        Mockito.when(channelPipeline.iterator()).thenReturn(iterator);
+        Mockito.when(iterator.hasNext()).thenReturn(true).thenReturn(true).thenReturn(true).thenReturn(false);
+        Mockito.when(iterator.next()).thenReturn(pipelineEntry);
+    }
+
+    @Test
+    public void testDecodeMessageSsl() throws Exception {
+        String expectedSubjectDN = "CN=Subject,O=Subject Organization";
+        String expectedIssuerDN = "CN=Issuer,O=Issuer Organization";
+
+        initSsl();
+
+        List<Object> decoded = new ArrayList<>(1);
+        decode(channelHandlerContext, "test".getBytes(), decoded);
+
+        X500Principal actualSubject = ((ByteArrayMessage)decoded.get(0)).getSslSessionStatus().getSubject();
+        X500Principal actualIssuer = ((ByteArrayMessage)decoded.get(0)).getSslSessionStatus().getIssuer();
+        assertEquals(expectedSubjectDN, actualSubject.getName());
+        assertEquals(expectedIssuerDN, actualIssuer.getName());
+    }
+
+    @Test
+    public void testDecodeMessageNoSsl() {
+        initNoSslHandler();
+
+        List<Object> decoded = new ArrayList<>(1);
+        decode(channelHandlerContext, "test".getBytes(), decoded);
+
+        SslSessionStatus sslSessionStatus = ((ByteArrayMessage)decoded.get(0)).getSslSessionStatus();
+        assertNull(sslSessionStatus);
+    }
+
+    @Test
+    public void testDecodeMessageNoCertificates() throws Exception {
+        initSslWithNoCertificates();
+
+        List<Object> decoded = new ArrayList<>(1);
+        decode(channelHandlerContext, "test".getBytes(), decoded);
+
+        SslSessionStatus sslSessionStatus = ((ByteArrayMessage)decoded.get(0)).getSslSessionStatus();
+        assertNull(sslSessionStatus);
+    }
+
+    @Test
+    public void testDecodeMessagePeerUnverified() throws Exception {
+        initSslPeerUnverified();
+
+        List<Object> decoded = new ArrayList<>(1);
+        decode(channelHandlerContext, "test".getBytes(), decoded);
+
+        SslSessionStatus sslSessionStatus = ((ByteArrayMessage)decoded.get(0)).getSslSessionStatus();
+        assertNull(sslSessionStatus);
+    }
+
+    private void initSsl() throws Exception {
+        initSslEngine();
+        try (InputStream inStream = new FileInputStream("src/test/resources/netty/client_certificate.cer")) {
+            CertificateFactory cf = CertificateFactory.getInstance("X.509");
+            certificate = (X509Certificate)cf.generateCertificate(inStream);
+        }
```

Comment:
Static certificate files should not be checked into version control as they will eventually expire and lead to test failures. See `TemporaryKeyStoreBuilder` for an approach to generating certificates for testing.